pkg/controller/plan/kubevirt.go

```diff
@@ -860,7 +859,7 @@ func (r *KubeVirt) UpdateVmByConvertedConfig(vm *plan.VMStatus, pod *core.Pod, s
 		return
 	}
 
-	url := fmt.Sprintf("http://%s:8080/ovf", pod.Status.PodIP)
+	url := fmt.Sprintf("http://%s:8080/firmware", pod.Status.PodIP)
```

Comment:
why changing it back to the firmware?
let's get @rwmjones's ack on our intention and if we're aligned with how it will be reported by virt-v2v in future versions, we could minimize the changes here

@rwmjones [when migrating to RHV, virt-v2v sets the bios-type that includes the firmware type in the generated ovf](https://github.com/libguestfs/virt-v2v/commit/c4d2800d90bf97118eef9a091a461126fad979a8), but MTV uses `-o local` so the generated OVF doesn't include it. would it make sense to add the firmware/bios-type also to `-o local`?

Reply:
Sighs .. we actually have an `-o kubevirt` mode already which is really what you should use, and fix if it doesn't work.

However about this question, `-o local` doesn't set the firmware but it probably should do.  As always, we need a bug to track this.

Reply:
@rwmjones Thank you for the feedback. I will open an issue to address this.

Initially, we explored the `-o kubevirt` option but found that it provides less information than the `-o local` option. Therefore, instead of using the generated YAML directly, we thought of extracting the missing information from the generated OVF file to use during VM creation.

Additionally, it's worth mentioning that our tests with the kubevirt option revealed it also does not include the firmware type field.

Reply:
The way we want this to work if that you tell us what's missing from `-o kubevirt` and we'll add it.  That makes virt-v2v better too because it means people can use it directly for small conversions.  What do you need and what is missing?

Reply:
I'm not entirely sure of the specifics, but @nyoxi  investigated this [issue](https://issues.redhat.com/browse/MTV-513) a while ago and found a gap between the provided configuration and our requirements.

With the proposed changes to the -o kubevirt fields, this could be a really interesting direction to explore. We can begin by adding the firmware type, as it's our most urgent need, and use it as a baseline for incorporating other missing configurations in the future.

Reply:
Thanks - please file bug(s) in Jira or upstream about any features you need.

Reply:
Thanks, just created this issue for beginning https://github.com/libguestfs/virt-v2v/issues/45

Reply:
Actually I should have said in Jira **and** upstream, since I guess you will want this to be in RHEL and without a Jira bug it won't get there.

Reply:
@bkhizgiy @rwmjones let me try to scope this down:
1. On the one extreme, letting virt-v2v to post the VM (that's an idea we had long time ago) is complicated because we need to provide a service account, possibly on a remote cluster and breaks the generalization in Forklift.
2. On the other extreme, we cannot ignore the output of virt-v2v as Forklift did so far - here we have a concrete case we need this information.

Forklift currently build the entire VM configuration and with this change take the firmware produced by virt-v2v - I think we should still return the entire configuration that virt-v2v produces. There are two options:
1. Get the OVF virt-v2v produces with `-o local`
2. Get a YAML virt-v2v produces with `-o kubevirt`

The OVF includes more properties but to solve this issue we only need the firmware.
I suggest that we'll change virt-v2v command to include `-o kubevirt` and return that YAML:
1. Logically, we migrate to KubeVirt...
2. We can simplify parsing in Forklift by creating a VirtualMachine from that input (rather than parsing OVF)
3. As we go, we can add stuff to the YAML (in virt-v2v) until we take that as a baseline and adjust it, if needed, based on input from the user

If you agree with me so far, how about letting the conversion pod in Forklift add the firmware configuration to the YAML until we upgrade virt-v2v to a version in which it does it by its own? There won't be much room for mistake because we'll follow the format that KubeVirt stores this property within the YAML (with `cnv.Bootloader{BIOS: &cnv.BIOS{}}` or `cnv.Bootloader{ EFI: &cnv.EFI{ SecureBoot: ptr.To(false)}}`)

Reply:
@ahadas Sounds good to me. It makes more sense to use the `-o kubevirt` option as we discussed yesterday, so if it's agreed, I'll make the suggested changes.

Reply:
Sounds good here too.